pkg/synopsysctl/cmd_update.go

```diff
@@ -386,6 +386,41 @@ func updateAlert(alt *alertapi.Alert, flagset *pflag.FlagSet) (*alertapi.Alert,
 	}
 	newSpec := alertInterface.(alertapi.AlertSpec)
 	newSpec.Environs = util.MergeEnvSlices(newSpec.Environs, alt.Spec.Environs)
+
+	// check whether the update Alert version is greater than or equal to 5.0.0
+	isGreaterThanOrEqualTo, err := util.IsNotDefaultVersionGreaterThanOrEqualTo(newSpec.Version, 5, 0, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	// if greater than or equal to 5.0.0, then copy PUBLIC_HUB_WEBSERVER_HOST to ALERT_HOSTNAME and PUBLIC_HUB_WEBSERVER_PORT to ALERT_SERVER_PORT
+	// and delete PUBLIC_HUB_WEBSERVER_HOST and PUBLIC_HUB_WEBSERVER_PORT from the environs. In future, we need to request the customer to use the new params
+	if isGreaterThanOrEqualTo {
+		isChanged := false
+		maps := util.StringArrayToMapSplitBySeparator(newSpec.Environs, ":")
+		_, ok := maps["PUBLIC_HUB_WEBSERVER_HOST"]
```

Comment:
I think we can simply this logic thusly:

if _, ok := maps["PUBLIC_HUB_WEBSERVER_HOST"]; ok {
  if _, ok1 := maps["ALERT_HOSTNAME"]; !ok1 {
    maps["ALERT_HOSTNAME"] = maps["PUBLIC_HUB_WEBSERVER_HOST"]
  }
  delete(maps, "PUBLIC_HUB_WEBSERVER_HOST")
  isChanged = true
}

Same logic for PORT

Then lastly re-do the conversion:
if isChanged {
  newSpec.Environs = util.MapToStringArrayJoinBySeparator(maps, ":")
}

Reply:
done